### ISSUE TITLE
Merge IIIF/OpenSeaDragon support

### ISSUE THREAD

**Issue by lucaro** [vitrivr/vitrivr-ng#14]
The OpenSeaDragon inclusion which enables support for IIIF and zoomable images in general, added in https://github.com/LukasStoeckli/vitrivr-ng/tree/dev should be merged here as well. Maybe some cleanup is necessary first. Any thoughts @silvanheller @ppanopticon ?

**Comment by silvanheller** [comment]
Well if OpenSeaDragon is truly general-purpose, i don't see why we shouldn't be able to replace our current image / imagesequence detail popup viewer with it. I'm not sure about the differentiation between IIIF / non-IIIF Images and why that is made - vitrivr-ng also loads 'normal' images via http?

# Detailed Comments

https://github.com/LukasStoeckli/vitrivr-ng/blob/d8cdf4caae555825545b4ddadd9cb39e481c897c/src/app/core/basics/resolver.service.ts#L79 and 
https://github.com/LukasStoeckli/vitrivr-ng/blob/d8cdf4caae555825545b4ddadd9cb39e481c897c/src/app/core/basics/resolver.service.ts#L107 
are ugly hacks - this needs more work to differentiate or integrate IIIF / non-IIIF URLs

https://github.com/LukasStoeckli/vitrivr-ng/blob/dev/src/app/objectdetails/quick-viewer.component.html

Is a plain demo-hack, but it looks like openseadragon can replace our generic image / imagesequenceviewer so this is a trivial change to merge

I don't know what this code snippet does: https://github.com/LukasStoeckli/vitrivr-ng/blob/d8cdf4caae555825545b4ddadd9cb39e481c897c/src/app/results/gallery/mini-gallery.component.ts#L147 - my guess is it just accesses the OpenSeaDragon Viewer? But that seems ok to merge

**Comment by ppanopticon** [comment]
I agree with Silvan here: The first three commits are quick & dirty hacks and require some serious rework before we can merge this with the UI. 

If I understand the code correctly, the implementation simply replaces the current media resolution mechanism, which is probably not what we want. Instead, we'd rather have the choice between IIIF and "ordinary" HTTP access to media e.g. via the config.

Given the extent of this addition and the very few commits in https://github.com/LukasStoeckli/vitrivr-ng/tree/dev, it's probably easier to just add IIIF support ourselves instead of cherrypicking parts of that code.

**Comment by lucaro** [comment]
It is also my understanding, that OpenSeaDragon is a viewer for zoomable images which supports zoomable images from various sources, including but not limited to IIIF. Having it as the default image viewer is probably overkill in many cases but it probably would not hurt either. We should maybe have a look at what other features it has which might be useful in our context, see https://openseadragon.github.io/

**Comment by silvanheller** [comment]
I've put a help wanted label on this - it's not a priority and a good issue for someone wanting to contribute to vitrivr-ng

**Comment by lucaro** [comment]
Agreed. Once the API on the cineast side is transitioned to javalin, we might also consider having the routes through which cineast serves content to be complient with at least the basic IIIF functionality.

**Comment by silvanheller** [comment]
I assume we can close this after gsoc? @Spiess @lucaro

**Comment by lucaro** [comment]
Since #84 is already merged, we can close this.
